File: app/fsharp/Power.fs
module Power

open System
open System.IO

type PowerInfo = {
    psu: string;
    status: string;
    level: Int32;
}

let info () = 
    let psu =
        if File.Exists("/sys/class/power_supply/BAT0/status")
            then "Cell"
            else "AC"
    let status = 
        if psu = "Cell"
            then Util.readLine("/sys/class/power_supply/BAT0/status")
            else "Unknown"
    let level = 
        if psu = "Cell"
            then Int32.Parse(Util.readLine("/sys/class/power_supply/BAT0/capacity"))
            else 0
    {psu=psu;status=status;level=level}

    

File: lib/power.ml
open Futil
open Fseerrec.Power

let info () = 
    let psu =
        if Sys.file_exists "/sys/class/power_supply/BAT0/status"
            then "Cell"
            else "Psu"
    in
    let status = 
        if psu = "Cell"
            then 
                match read_file_line "/sys/class/power_supply/BAT0/status" with
                | Some x -> x
                | None -> "Unknown"
            else "Unknown"
    and
        level = 
        if psu = "Cell"
            then 
                match read_file_line "/sys/class/power_supply/BAT0/capacity" with
                | Some x -> Int32.of_string x
                | None -> 0l
            else 0l
    in
    {psu=psu;status=status;level=Int32.to_int level}

    
